### ISSUE TITLE
Changing options on cloned TcpStream doesn't propagated to the original one

### ISSUE THREAD

**Issue by Numenorean** [rust-lang/rust#95336]
I tried this code:

```rust
use std::{net::TcpStream, time::Duration};

fn main(){
    let stream = TcpStream::connect("google.com:80").unwrap();
    let copy = stream.try_clone().unwrap();

    println!("{:?}", stream.read_timeout());
    copy.set_read_timeout(Some(Duration::from_millis(500)));
    println!("{:?}", stream.read_timeout());
}
```

I expected to see this happen: the original stream timeout also is changed by modifing the copy
Instead, this happened: it seems TcpStream copy doesn't inherits specific options

On beta and nightly the same situation

### Meta

`rustc --version --verbose`:
```
rustc 1.59.0 (9d1b2106e 2022-02-23)
binary: rustc
commit-hash: 9d1b2106e23b1abd32fce1f17267604a5102f57a
commit-date: 2022-02-23
host: x86_64-pc-windows-msvc
release: 1.59.0
LLVM version: 13.0.0
```

<details><summary>Backtrace</summary>
<p>

```
<backtrace>
```

</p>
</details>

**Comment by sfackler** [comment]
This is up to the operating system.

**Comment by bugadani** [comment]
> This is up to the operating system.

Maybe, but the [docs](https://doc.rust-lang.org/stable/std/net/struct.TcpStream.html#method.try_clone) say the following:

> The returned TcpStream is a reference to the same stream that this object references. Both handles will read and write the same stream of data, and **options set on one stream will be propagated to the other stream**.

**Comment by Numenorean** [comment]
I agree with @bugadani, if some feature doesn't work because of the system on which you are running the code, it must be specified in docs, because now people are excpecting it works as discribed in the docs (and me too by the way), it's a little bit confusing

**Comment by hkBst** [comment]
@Numenorean, like @sfackler said, the TcpStream wraps a Socket which is platform-specific, and for example on Linux wraps a FileDesc, while on Windows it wraps an OwnedSocket which wraps a ValidRawSocket which is a NotAllOnes<RawSocket> where a RawSocket is a u32 or u64 depending on your platform.

@ChrisDenton since this may be a Windows-specific issue.
